### PR TITLE
added linting on R code (fixes #179)

### LINKED ISSUE #179: Add linting on R code
We should use a linter to somewhat-standardize the style of the R code in this project. For R code, I'm fond of the `covr` package.

To close this issue, introduce linting in the tests run on Travis. This issue will involve reading our `.travis.yml` and the scripts in `.ci/` to get an understanding of how `uptasticsearch`'s Continuous Integration (CI) setup.

I recommend taking an approach similar to the one in microsoft/LightGBM#2437, but it's up to you (the contributor) to propose how you want to handle it.

This is a medium-difficulty issue, but one you can do without understanding anything about Elasticsearch or `uptasticsearch`'s R code.

### PR DESCRIPTION
**Pull request by jameslamb**
This PR introduces a linter for the R package, using  `{lintr}`.

### CODE REVIEW

File: .ci/lint_r_code.R [hunk 1]
```diff
@@ -0,0 +1,109 @@
+
+library(lintr)
+
+args <- commandArgs(
```
Comment (austin3dickey) [inline .ci/lint_r_code.R:4]: TIL `commandArgs`. Always better to use base R and reduce dependencies if it's easy. :)

File: r-pkg/R/es_search.R [hunk 2]
```diff
@@ -737,8 +737,8 @@ es_search <- function(es_host
 # [examples]
 # \dontrun{
 # #--- Example 1: Basic Usage ---#
-# .ConvertToSec('1h') # returns 60*60 = 3600
-# .ConvertToSec('15m') # returns 60*15 = 900
+# .ConvertToSec('1h') # returns 60 * 60 = 3600
+# .ConvertToSec('15m') # returns 60 * 15 = 900
```
Comment (austin3dickey) [inline r-pkg/R/es_search.R:741]: interesting, it lints comments too!

File: .ci/test.sh [hunk 3]
```diff
@@ -4,6 +4,7 @@
 set -e
 
 if [[ "$TASK" == "rpkg" ]]; then
+  Rcript .ci/lint_r_code.R $(pwd)
```
Comment (austin3dickey) [inline .ci/test.sh:7]: I think you meant "Rcrypt" ☠️ 
```suggestion
  Rscript .ci/lint_r_code.R $(pwd)
```

File: r-pkg/R/es_search.R (hunk 2 above)
Comment (jameslamb) [reply r-pkg/R/es_search.R:741]: ha it doesn't, but when I did find-in-all-files for `*6` these  showed up and I thought "eh might as well"

File: .ci/test.sh (hunk 3 above)
Comment (jameslamb) [reply .ci/test.sh:7]: ha yup that would do it

**Pull request merged**